lib/orderbook/OrderBook.ts

```diff
@@ -87,16 +87,24 @@ class OrderBook extends EventEmitter {
     }
   }
 
+  private persistOrderById = ({ orderId, pairId, peerPubKey }: { orderId: string, pairId: string, peerPubKey?: string }) => {
+    const order = peerPubKey ? this.getPeerOrder(orderId, pairId, peerPubKey) : this.getOwnOrder(orderId, pairId);
+    return this.repository.addOrderIfNotExists(order);
+  }
+
   private bindSwaps = () => {
     if (this.swaps) {
-      this.swaps.on('swap.paid', (swapResult) => {
+      this.swaps.on('swap.paid', async (swapResult) => {
+        await this.persistOrderById(swapResult);
```

Comment:
This was buggy per @erkarl's crash report so I've changed it some already if you want to take another look, but to answer your questions:

1. Yes this was indeed the bug and was fixed I believe.

2. Basically, if it's our own maker order, we won't even know about the peer order. All we know is a peer wanted to swap with us. If it's a peer maker order, I figure we will want to persist our own taker order, but I'm thinking that should happen as part of saving the trade to the db. For this PR I was mainly just focusing on db schema/structure and not breaking any existing functionality.

3. Yes we do, but that would be same as 2 above where that should happen when we persist the trade, which we're not doing yet, and can be handled in a future PR.

4. No particular reason, I don't think the order matters here though.

Reply:
Actually for 4, the order does matter, because we want to persist the order before we remove it. After we've removed it, we won't have the information any longer.

Reply:
I continued the discussion in the main thread, since you already were commented about some of these things.